syntax = "proto3";

message Centroid {
    int32 idx = 1;
    double x = 2;
    double y = 3;
}

message Cent {
    double x = 1;
    double y = 2;
}


message ShuffleSortRequest {
    int32 centroidID = 1;
    int32 numMappers = 2;
    repeated int32 mapAddresses = 3;
}

message ReducerResponse {
    int32 centroidID = 1;
    Cent updatedCentroids = 2;
}

service Reducer {
    rpc ShuffleAndSort(ShuffleSortRequest) returns (ReducerResponse);
}